cmd/edit/cluster/cmd.go

```diff
@@ -1142,3 +1189,163 @@ func confirmMigration() (bool, error) {
 		Default: false,
 	})
 }
+
+// AutoNode helper functions
+func autonodeExists(cluster *cmv1.Cluster) bool {
+	return cluster.AutoNode() != nil && cluster.AutoNode().Mode() == "enabled"
+}
+
+func autonodeRoleExists(cluster *cmv1.Cluster) string {
+	if cluster.AWS() != nil && cluster.AWS().AutoNode() != nil {
+		return cluster.AWS().AutoNode().RoleArn()
+	}
+	return ""
+}
+
+func setAutoNode(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.Cluster, autonode string, roleArn string) (
+	mode string, role string, err error) {
+
+	// Check if flags are changed
+	autonodeChanged := cmd.Flags().Changed("autonode")
+	roleArnChanged := cmd.Flags().Changed("autonode-iam-role-arn")
+
+	if !autonodeChanged && !roleArnChanged {
+		return "", "", nil
+	}
+
+	// Validate HCP cluster
+	if !aws.IsHostedCP(cluster) {
+		return "", "", fmt.Errorf("AutoNode is only supported for Hosted Control Plane clusters")
+	}
+
+	// Get current state
+	currentAutoNodeEnabled := autonodeExists(cluster)
+	currentRoleArn := autonodeRoleExists(cluster)
+
+	// Validate autonode value
+	if autonodeChanged {
+		if autonode != "enabled" {
+			return "", "", fmt.Errorf("Invalid value for --autonode. Currently only 'enabled' is supported")
+		}
+		mode = autonode
+	}
+
+	// Handle role ARN
+	if roleArnChanged {
+		if roleArn != "" && !aws.RoleArnRE.MatchString(roleArn) {
+			return "", "", fmt.Errorf("Invalid IAM Role ARN format: %s", roleArn)
+		}
+		role = roleArn
+	}
+
+	// If enabling AutoNode, role ARN is required
+	if mode == "enabled" && !currentAutoNodeEnabled {
+		if role == "" && currentRoleArn == "" {
+			return "", "", fmt.Errorf("IAM Role ARN is required when enabling AutoNode. Use --autonode-iam-role-arn to specify the role")
+		}
+		if role == "" {
+			role = currentRoleArn
+		}
+	}
+
+	// If only updating role ARN, AutoNode must already be enabled
+	if roleArnChanged && !autonodeChanged && !currentAutoNodeEnabled {
+		return "", "", fmt.Errorf("Cannot update AutoNode IAM Role ARN when AutoNode is not enabled. Enable AutoNode first using --autonode=enabled")
+	}
+
+	// If updating role ARN with AutoNode already enabled, maintain enabled state
+	if roleArnChanged && currentAutoNodeEnabled && !autonodeChanged {
+		mode = "enabled"
+	}
+
+	// Confirm with user
+	if mode != "" || role != "" {
+		confirmAutoNode(r, mode, role, currentAutoNodeEnabled, currentRoleArn)
+	}
```

Comment:
Curious- should role ever be empty if mode isn't with the validations above?

Reply:
fixed on last commit

Reply:
@olucasfreitas Did you test it? Did it solve anything?